addons/spectrum/functions/fnc_drawSpectrumLine.sqf

```diff
@@ -0,0 +1,56 @@
+#include "script_component.hpp"
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fnc_drawSpectrumLine.sqf
+Parameters: colour - string corresponding to a colour in CfgMarkerColors
+Return: 
+
+Script to automatically draw a line on the map in the direction they are facing, labelled with the current freq. of the Spectrum Device
+Player must have the Spectrum Device equipped, be in "GUNNER" (i.e. zoomed-in) view, and have a GPS (or UAV terminal) assigned
+
+*///////////////////////////////////////////////
+
+params [["_colour", "ColorBlack", [""]]];
+
+if(!(
+		GVAR(spectrumAutoline) &&
+		(currentWeapon player) isKindOf ["hgun_esd_01_F", configFile >> "CfgWeapons"] &&
+		(
+			(player getSlotItemName 612) isKindOf ["ItemGPS", configFile >> "CfgWeapons"] ||
+			(player getSlotItemName 612) isKindOf ["UavTerminal_base", configFile >> "CfgWeapons"]
+		) &&
+		cameraOn == player && cameraView == "GUNNER"
+	)
+) exitWith { false };
+
+if(!isClass (configFile >> "CfgMarkerColors" >> _colour)) then { _colour = "ColorBlack"; };
+
+private _freq = ((missionnamespace getVariable ["#EM_SelMin", 141.6]) + (missionnamespace getVariable ["#EM_SelMax", 141.9]))/2;
+_freq = (round (_freq*10))/10;
+
+private _marker = createMarkerLocal ["_USER_DEFINED"+(getPlayerUID player)+str(getPos player)+str(getDir player)+str(_freq), player, currentChannel, player];
+_marker setMarkerShapeLocal "polyline";
+_marker setMarkerColorLocal _colour;
+
+
+private _l = 10000; // TODO: base on size of map?
```

Comment:
There is a few functions that can help you with getting map-size. Or grid-size and base it on x grids. Although hardcoded is fine too, as if it feels fitting, then you could redraw or continue drawing as you have a straight line to follow.

If you decide to base it on map size you need to test it with both small maps and maps like Kidal to ensure it works fine. I would also only get the world size once in ``postInit`` and cache it for use in the function. To avoid having to recalculate it every time. 

https://community.bistudio.com/wiki/worldSize  
https://community.bistudio.com/wiki/BIS_fnc_mapGridSize  
https://community.bistudio.com/wiki/BIS_fnc_mapSize

Reply:
As the max range of `Set Spectrum Signal Source` is 5000, leaving the line size hardcoded as 10000 is probably fine for now; but can change later if it causes issues

Reply:
That works